File: ELearningBackEnd/src/main/java/com/example/elearningbackend/discount/DiscountReq.java
package com.example.elearningbackend.discount;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
public class DiscountReq {

    private String code;

    private String type;

    private float discount;

    @JsonFormat(pattern = "dd-MM-yyyy")
    private LocalDate validFrom;

    @JsonFormat(pattern = "dd-MM-yyyy")
    private LocalDate validTo;

    private Integer quantity;

    private Long courseId;

    private boolean isActivated;
}

File: ELearningFrontEnd/src/app/dtos/discount/discount.req.ts
export class DiscountReq{


    public code: string;
    public type: string;
    public discount: number;
    public validFrom: Date;
    public validTo: Date;
    public quantity: number;
    public courseId: number | null;
    public activated: boolean
    constructor(data: any) {

        this.code = data.code;
        this.type = data.type;
        this.discount = data.discount;
        this.validFrom = data.validFrom;
        this.validTo = data.validTo;
        this.quantity = data.quantity;
        this.courseId = data.courseId;
        this.activated = data.activated;
    }

}